src/ray/core_worker/core_worker.cc

```diff
@@ -611,6 +611,13 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
   // Verify driver and worker are never mixed in the same process.
   RAY_CHECK_EQ(options_.worker_type != WorkerType::DRIVER, niced);
 #endif
+
+  // Notify that core worker is initialized.
+  {
+    absl::MutexLock lock(&initialize_mutex_);
+    initialized_ = true;
```

Comment:
Should we have some timeout here to avoid infinite hang? Maybe sth like 30 seconds? (and we can probably check fail if it is not done on time?)

Reply:
moved this to a scope_guard to ensure it will eventually be called.